### PR TITLE
Build NUnit.ConsoleRunner.NetCore as a .NET Core Tool.

### PR DESCRIPTION
**Pull request by ejball**
I believe this is all that is necessary to build the NuGet package as a .NET Core Tool. It seemed to install and work fine in my (limited) local testing.

* https://docs.microsoft.com/en-us/dotnet/core/tools/global-tools
* https://natemcmaster.com/blog/2018/05/12/dotnet-global-tools/

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/nunit/nunit-console?pullRequest=810) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/nunit/nunit-console?pullRequest=810) before we can accept your contribution.<br/><br/>:x: ejball [sign now](https://cla.dotnetfoundation.org/nunit/nunit-console?pullRequest=810)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/nunit/nunit-console?pullRequest=810) it.</sub>

File: nuget/runners/DotnetToolSettings.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<DotNetCliTool>
+  <Commands>
+    <Command Name="nunit3-console" EntryPoint="nunit3-console.dll" Runner="dotnet" />
```
Comment (rprouse) [inline nuget/runners/DotnetToolSettings.xml:4]: This is a chance to simplify the command. Should we use something simpler and easy to type? Maybe just `nunit`? @ChrisMaddock ?

**Comment by ChrisMaddock** [comment]
Hey @ejball - thanks for putting this together. As discussed on the #475 - I'm a little concerned about how possible it's going to be to do this in a way which is accurate in all cases - I'd like to understand more about why XUnit deprecated this approach when the did.

@bradwilson - are you able to help us understand why XUnit chose to deprecate `dotnet xunit`? I've had a bit of a Google, but struggling to find much info on this.

File: nuget/runners/DotnetToolSettings.xml (hunk 1 above)
Comment (ChrisMaddock) [reply nuget/runners/DotnetToolSettings.xml:4]: Yep - definitely `dotnet nunit` is what we want, if we choose to go ahead with this. (See comment below!)

File: nuget/runners/nunit.console-runner.netcore.nuspec [hunk 2]
```diff
@@ -22,32 +22,36 @@
...
+    <file src="bin/netcoreapp3.1/nunit.engine.api.pdb" target="tools/netcoreapp3.1/any" />
+    <file src="bin/netcoreapp3.1/nunit.engine.api.xml" target="tools/netcoreapp3.1/any" />
+    <file src="bin/netcoreapp3.1/Microsoft.DotNet.InternalAbstractions.dll" target="tools/netcoreapp3.1/any" />
+    <file src="bin/netcoreapp3.1/Mono.Cecil.dll" target="tools/netcoreapp3.1/any" />
+    <file src="bin/netcoreapp3.1/Mono.Cecil.Mdb.dll" target="tools/netcoreapp3.1/any" />
+    <file src="bin/netcoreapp3.1/Mono.Cecil.Pdb.dll" target="tools/netcoreapp3.1/any" />
+    <file src="bin/netcoreapp3.1/Mono.Cecil.Rocks.dll" target="tools/netcoreapp3.1/any" />
+    <file src="bin/netcoreapp3.1/System.Xml.XPath.XmlDocument.dll" target="tools/netcoreapp3.1/any" />
+    <file src="../../nuget/runners/nunit.console.nuget.addins" target="tools/netcoreapp3.1/any"/>
+    <file src="../../nuget/runners/DotnetToolSettings.xml" target="tools/netcoreapp3.1/any"/>
```
Comment (ChrisMaddock) [inline nuget/runners/nunit.console-runner.netcore.nuspec:54]: Regardless of what we decide to do with this particular issue - this raises a good point. We should migrate the files to this dir regardless, so we can change our minds in future. Thanks!

**Comment by bradwilson** [comment]
> @bradwilson - are you able to help us understand why XUnit chose to deprecate `dotnet xunit`? I've had a bit of a Google, but struggling to find much info on this.

I was unable to reliably deal with assembly resolution issues on .NET Core. My v3 architecture shift attempts to address this in a different way: unit tests projects in xUnit.net v3 will be themselves stand-alone executables (EXE files for .NET Framework, and `dotnet run`nable projects for .NET Core). Solves all the assembly resolution issues, as well as the version collision issues that caused me to stop taking any external dependencies.

**Comment by CharliePoole** [comment]
@bradwilson Interesting. As it turns out, that's what I'm doing for TCLite right now.

**Comment by ChrisMaddock** [comment]
Thanks @bradwilson - appreciate that! Good to know that the problem wasn't with the nature of .NET Core Tools themselves so much, more the whole idea of a separate runner. I'd had thought you'd just deprecated `dotnet xunit`, but were still supporting a separate test runner.

That...probably gives us a lot to think about! NUnit has self-executing test assemblies as you're probably aware, but they're much for the simplest of solutions, so don't allow access to all the functionality that's currently in the console/engine layer.

**Comment by bradwilson** [comment]
It's still my intention to ship stand-alone runners (`xunit.console.exe`, for example), for two reasons: (a) they support backward compatibility, so that's the only way to run v1 and v2 tests; and (b) that's the only way to run multiple assemblies in parallel. It's just that when running v3 assemblies, they'll be in a separate process rather than in-process.

**Comment by ChrisMaddock** [comment]
Ahh, that makes sense. We've had conversations about removing in-process running in NUnit as well, although for completely different reasons. Good to know that still seems like a viable option, although we've got a fair bit of work to do to get there on our side.

Thanks for your advice here!

File: nuget/runners/DotnetToolSettings.xml (hunk 1 above)
Comment (ChrisMaddock) [inline nuget/runners/DotnetToolSettings.xml:4]: Thanks for making this change @ejball! I'd happy enough that the Core Tool shouldn't be an issue here - more other problems. Would you just be able to change the tool name please, as we can merge this in?

Thanks for the contribution!

```suggestion
    <Command Name="nunit3" EntryPoint="nunit3-console.dll" Runner="dotnet" />
```

**Comment by ChrisMaddock** [comment]
Thanks for doing this @ejball - sorry, had missed the notification for you making the last change. I've just resolved conflicts and will merge when CI passes.

Thanks! 😄

File: nuget/runners/nunit.console-runner.netcore.nuspec [hunk 3]
```diff
@@ -22,29 +22,33 @@
...
-    <file src="bin/netcoreapp3.1/nunit.engine.api.pdb" target="tools" />
-    <file src="bin/netcoreapp3.1/nunit.engine.api.xml" target="tools" />
-    <file src="bin/netcoreapp3.1/Microsoft.DotNet.InternalAbstractions.dll" target="tools" />
-    <file src="bin/netcoreapp3.1/testcentric.engine.metadata.dll" target="tools" />
-    <file src="bin/netcoreapp3.1/System.Xml.XPath.XmlDocument.dll" target="tools" />
-    <file src="../../nuget/runners/nunit.console.nuget.addins" target="tools"/>
+    <file src="bin/netcoreapp3.1/nunit3-console.exe" target="tools/netcoreapp3.1/any" />
+    <file src="bin/netcoreapp3.1/nunit3-console.pdb" target="tools/netcoreapp3.1/any" />
+    <file src="bin/netcoreapp3.1/nunit3-console.dll" target="tools/netcoreapp3.1/any" />
+    <file src="bin/netcoreapp3.1/nunit3-console.dll.config" target="tools/netcoreapp3.1/any />
```
Comment (ChrisMaddock) [inline nuget/runners/nunit.console-runner.netcore.nuspec:36]:
```suggestion
    <file src="bin/netcoreapp3.1/nunit3-console.dll.config" target="tools/netcoreapp3.1/any" />
```

Missed a quote with the merge conflict

File: nuget/runners/DotnetToolSettings.xml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<DotNetCliTool>
+  <Commands>
+    <Command Name="nunit3" EntryPoint="nunit3-console.dll" Runner="dotnet" />
```
Comment (ChrisMaddock) [inline nuget/runners/DotnetToolSettings.xml:4]:
```suggestion
    <Command Name="nunit" EntryPoint="nunit3-console.dll" Runner="dotnet" />
```

Rename tools as dotnet nunit - error in my own previous suggestion...

**Review by ChrisMaddock — APPROVED** [review]

**Comment by ChrisMaddock** [comment]
For some reason Azure hadn't registered this PR. Closed and Re-opened, CI is running now.

**Comment by ChrisMaddock** [comment]
Thanks @ejball!

**Comment by ejball** [comment]
Will this be published to nuget.org?

**Pull request merged**